geth/api/api.go

```diff
@@ -58,7 +58,17 @@ func (api *StatusAPI) StartNode(config *params.NodeConfig) error {
 // StartNodeAsync start Status node, fails if node is already started
 // Returns immediately w/o waiting for node to start (see node.ready)
 func (api *StatusAPI) StartNodeAsync(config *params.NodeConfig) <-chan error {
-	return runAsync(func() error { return api.StartNode(config) })
+	if !api.b.IsNodeRunning() {
+		return runAsync(func() error { return api.StartNode(config) })
+	}
+	return runAsync(func() error {
+		node, err := api.b.NodeManager().Node()
```

Comment:
That looks dangerous. if someone calls `StartNodeAsync` twice, it will wait forever.
We probably need some state like `starting/stopping` to handle that gracefully.

Reply:
i started writing similar thing but then decided that it doesn't need to be that complex and it is better to get rid from such api at all in a followup with status-react

instead introduce a single function, like ensureNodeStart(config) or even change signature of restart node to accept a config

Reply:
> change signature of restart node to accept a config
that is a good idea, IMO